Fix double-counting of alignment offset in `arena_alloc_aligned`

logmoon opened the pull request:
## Summary
Fixes the bug reported in issue #19 where `arena_alloc_aligned` incorrectly reports out-of-memory errors due to double-counting the alignment offset.

## Problem
The function calculates an alignment `offset` and adjusts `arena->index` to the aligned position. However, the space check then adds `offset` to the already-adjusted `arena->index`, effectively counting the alignment padding twice.

**Before:**
```c
if (arena->size - (arena->index + offset) < size) {
    return NULL;
}
```

**After:**
```c
if (arena->size - arena->index < size) {
    return NULL;
}
```

## Changes
- Fixed the space check in `arena_alloc_aligned` to use the already-adjusted `arena->index`
- Added test case `arena_alloc_aligned_tests.edge_case_tight_space` to verify the fix

## Testing
The new test covers the edge case where:
- Alignment adjustment is needed (offset > 0)
- Remaining space is tight but sufficient
- Previously would incorrectly return NULL

Example from test:
- Arena size: 30 bytes
- Current index: 10
- Requested: 14 bytes with 16-byte alignment
- Expected: Align to 16, allocate 14 bytes (total 30)
- Before fix: Would return NULL

All tests pass under valgrind.

ccgargantua reviewed (changes requested):
What's going on with this diff? All files changed. Line endings? Please resolve so that only your relevant changes are included in the commit.

logmoon committed:
Fix double-counting of alignment offset in space check

- Removed offset from space calculation since arena->index already adjusted
- Added test case for edge case where alignment causes tight space
- Fixes issue #19

logmoon commented:
Apologies for the line ending mess, I somehow completely missed it.
It's Fixed now and the diff should only show the bug fix and test addition.

ccgargantua commented on tests.c:
```diff
@@ -47,3 +47,21 @@ TEST(arena_alloc_tests, basic)
+	char region[30];
+	Arena arena;
+	arena_init(&arena, region, 30);
+	arena.index = 10;  // Start at offset 10
```
Why start at 10 in a region of 30 bytes rather than start at 0 in a region of 20 bytes?

ccgargantua commented on tests.c:
```diff
@@ -47,3 +47,21 @@ TEST(arena_alloc_tests, basic)
+	// Should align to index 16, then allocate 14 bytes (uses exactly 30 total)
+	void *ptr = arena_alloc_aligned(&arena, 14, 16);
+
+	// Should succeed, not return NULL (this was the bug)
```
Unnecessary comment here and on line 65. Remove those please. Oddly specific comment smells a bit like AI. If you are using AI, that's fine, but the code needs to be clean and fit the style of the rest of the code.

ccgargantua reviewed (changes requested):
One minor change requested, and I asked for an explanation on another piece of code. Both in the unit test you provided. Otherwise, change looks good.

logmoon commented:
I've removed the comments to match the style, I just wanted to explain what's happening and the expected behavior with the comments. No AI was used.

Regarding starting at index 10: this ensures we trigger the alignment adjustment (aka getting `offset > 0`) which is necessary to reproduce the bug where the space check double-counts the offset.
If we start at index 0 with 20 bytes, we don't need to adjust for alignment, we're already aligned, the offset would be 0 and we won't catch the bug.

ccgargantua commented:
> I've removed the comments to match the style, I just wanted to explain what's happening and the expected behavior with the comments. No AI was used.

Hope I didn't offend by asking 😅 I recently had a negative experience with a user submitting a PR entirely generated by AI. I'm overly cautious now.
